test/testers.jl

```diff
@@ -42,9 +42,17 @@ end
 
 @testset "testers.jl" begin
     @testset "test_scalar" begin
-        double(x) = 2x
-        @scalar_rule(double(x), 2)
-        test_scalar(double, 2.0)
+        @testset "happy path" begin
+            double(x) = 2x
+            @scalar_rule(double(x), 2)
+            test_scalar(double, 2.1)
+        end
+        @testset "unhappy path" begin
+            alt_double(x) = 2x
+            @scalar_rule(alt_double(x), 3)  # this is wrong, on purpose
+            failures = metatest_get_failures(()->test_scalar(alt_double, 2.1))
+            @test !isempty(failures)
```

Comment:
Should this, and the ones below do more than just `@test !isempty(failures)`?
Like maybe test that the failure is the one we expect?
Or is this good for now?
I worry testing exact failure condition might be too brittle.

Reply:
i think this is fine for now :)

Reply:
Cool, can always make it tighter if something slips through later